src/main/java/org/openstreetmap/atlas/geography/atlas/raw/creation/RawAtlasGenerator.java

```diff
@@ -300,6 +296,120 @@ private void populateAtlasMetadata()
+                if (pointsNeedingSyntheticTag.contains(identifier))
+                {
+                    // Add the synthetic tag
+                    tags.put(SyntheticDuplicateOsmNodeTag.KEY,
+                            SyntheticDuplicateOsmNodeTag.YES.toString());
+                }
+
+                // Add the Point
+                builder.addPoint(identifier, point.getLocation(), tags);
+            }
+        });
+
+        // Add Lines
+        atlas.lines().forEach(line ->
+        {
+            builder.addLine(line.getIdentifier(), line.asPolyLine(), line.getTags());
+        });
+
+        // Add Relations
+        atlas.relations().forEach(relation ->
+        {
+            final RelationBean bean = new RelationBean();
+            relation.members().forEach(member -> bean.addItem(member.getEntity().getIdentifier(),
+                    member.getRole(), member.getEntity().getType()));
+            builder.addRelation(relation.getIdentifier(), relation.getOsmIdentifier(), bean,
+                    relation.getTags());
+        });
+
+        // Build and return the new Atlas
+        return builder.get();
+    }
+
+    /**
+     * We may need to remove {@link Point}s from the built raw Atlas. There are two scenarios for
+     * removal:
+     * <p>
+     * <ul>
+     * <li>1. A {@link Point} was a shape point for an OSM {@link Way} that was removed. This point
+     * doesn't have any tags, isn't a part of a {@link Relation} and doesn't intersect with any
+     * other features in the Atlas.
+     * <li>2. A {@link Point} has multiple points at its {@link Location}. In this case, we sort all
+     * the Points, keep the one with the smallest identifier, add a
+     * {@link SyntheticDuplicateOsmNodeTag} and remove the rest of the duplicate Points. Note: we
+     * can potentially be tossing out OSM Nodes with non-empty tags. However, this is the most
+     * deterministic and simple way to handle this. The presence of the synthetic tag will make it
+     * easy to write an Atlas Check to resolve the data error.
+     * </ul>
+     *
+     * @param atlas
+     *            The {@link Atlas} to remove the Points from
+     * @return a new {@link Atlas} without the extra points or the given Atlas if no removal is
+     *         needed
+     */
+    private Atlas removeDuplicateAndExtraneousPointsFromAtlas(final Atlas atlas)
+    {
+        final Set<Long> pointsToRemove = new HashSet<>();
+        final Set<Long> duplicatePointsToKeep = new HashSet<>();
+
+        for (final Point point : atlas.points())
+        {
+            final Set<Long> duplicatePoints = Iterables.stream(atlas.pointsAt(point.getLocation()))
+                    .map(Point::getIdentifier).collectToSet();
+            if (!duplicatePoints.isEmpty() && duplicatePoints.size() > 1)
+            {
+                // Sort the points
+                final Set<Long> sortedDuplicates = Iterables.asSortedSet(duplicatePoints);
```

Comment:
I haven't seen more than 2-3 OSM Nodes stacked on top of each other. It could be possible if they were created programmatically and uploaded to OSM, but then I think we have bigger problems.